app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :expenses, foreign_key: 'author_id', dependent: :destroy
+  has_many :groups, foreign_key: 'user_id', dependent: :destroy
+
+  validates :name, :role, presence: true
```

Comment:
- [OPTIONAL] Great job adding user name validation 👏🏻 Let's now validate all user input (password, ...) to make sure that anyone with bad intentions cannot compromise their app.

Reply:
```rb
class User < ApplicationRecord
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable

  has_many :expenses, foreign_key: 'author_id', dependent: :destroy
  has_many :groups, foreign_key: 'user_id', dependent: :destroy

  validates :name, :role, :email, :password, presence: true

  def admin?
    role == 'admin'
  end
end
```